### ISSUE TITLE
rxjs/finnish counter-intuitive for functions assigned to variables/properties/params

### ISSUE THREAD

**Issue by jimivdw** [cartant/eslint-plugin-rxjs#105]
Given the following configuration:
```json
{
    "rxjs/finnish": [
        "error",
        {
            "strict": true
        }
    ]
}
```

We would currently get the following results:

```ts
function fun$(): Observable<never> { return EMPTY; }   // OK
const var$ = (): Observable<never> => EMPTY;           // ERROR shouldNotBeFinnish


class Cls {
  prop$ = (): Observable<never> => EMPTY;              // ERROR shouldNotBeFinnish
  mthd$(): Observable<never> { return EMPTY; }         // OK
  mthd2(par$: () => Observable<never>): void {}        // ERROR shouldNotBeFinnish
}

const var2$ = fun$;                                    // ERROR shouldNotBeFinnish
const var3$ = new Cls().mthd$;                         // ERROR shouldNotBeFinnish
```

To me, this feels counterintuitive. If a function/method should be in finnish notation if it returns an `Observable`, the same should be true when it is disguised as a variable/property/param.

Am I correct here? Or is this just not how the rule is meant to work.

Thanks in advance! And thanks in general to @cartant for maintaining this nice set of lint rules.

**Comment by JasonWeinzierl** [comment]
This is fixed in fork `eslint-plugin-rxjs-x` v0.9.0 .  Each of those examples should now be allowed by "strict".  Note this fix actually revealed a ton of Finnish violations in my codebase 🎉
